test-data/unit/check-parameter-specification.test

```diff
@@ -1092,3 +1092,20 @@ def func(callback: Callable[P, str]) -> Callable[P, str]:
         return 'baz'
     return inner
 [builtins fixtures/paramspec.pyi]
+
+[case testUnpackingParamsSpecArgsAndKwargs]
+from typing import Callable
+from typing_extensions import ParamSpec
+
+P = ParamSpec("P")
+
+def func(callback: Callable[P, str]) -> Callable[P, str]:
+    def inner(*args: P.kwargs, **kwargs: P.kwargs) -> str:
```

Comment:
`*args: P.kwargs` <- mypy should be complaining about this, no?

Reply:
Wow, um, yes, it should :)

Reply:
```suggestion
    def inner(*args: P.args, **kwargs: P.kwargs) -> str:
```

Reply:
Want to fix it / create an issue to follow up on it? :-)

Reply:
Definitely on (2), strong maybe on (1) :-)

Reply:
I'm also confused how unpacking the kwargs worked for you. Maybe it just got Any-ed?

Reply:
> I'm also confused how unpacking the kwargs worked for you. Maybe it just got Any-ed?

I think that bit's actually probably okay? I _think_ what happened is mypy inferred `ParamSpecArgs` for `*args` -- i.e., it "did what I wanted it to" even though my annotation was incorrect. It just didn't spot that `*args` has to be annotated with `P.args` in order for it to be a valid use of `ParamSpec`.

Reply:
I've created
- #13460